### PR TITLE
Add dynamic range

### PR DESCRIPTION
**Pull request by gjnoonan**
This patch adds Dynamic Range, options.based on the work of by Lowpro on the forums 

http://www.invelos.com/Forums.aspx?task=viewtopic&topicID=477881&PageNum=4&messageID=3073933#M3073933

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -2347,6 +2347,22 @@ function ProcessChildrenOf($boxparent, &$sql, &$numthispage, &$secondcol, &$thir
 		}
 		$dvd['format'] .= ' ' . $colours;
 
+		if ($dvd['format'] != '') {
```
Comment (julmud) [inline index.php:2350]: This block is useless. We know for a fact that `$dvd['format']` is not empty (it has at least a space, due to line 2348).

File: index.php [hunk 2]
```diff
@@ -2347,6 +2347,22 @@ function ProcessChildrenOf($boxparent, &$sql, &$numthispage, &$secondcol, &$thir
...
+		}
+
+		$dynamicRange = ''
+			. (($dvd['drhdr10']==1)	? "($lang[HDR10])": '')
+			. (($dvd['drdolbyvision']==1)	? "($lang[DOLBYVISION])": '');
+
+		if ($dvd['drhdr10']==1 && $dvd['drdolbyvision']==1) {
+			$dynamicRange = "($lang[HDR10], $lang[DOLBYVISION])";
+		}
+		if ($dynamicRange == '') {
```
Comment (julmud) [inline index.php:2361]: If `$dynamicRange` is empty, there's no need to concatenate it to `$dvd['format']` (line 2362 is in fact only adding two empty strings to `$dvd['format']`

**Review by julmud — COMMENTED** [review]
As there's a change of the DB schema structure, the schema version should be updated in schema.sql and schema.pre-4.1.sql (`db_schema_version`) and in global.php (`$code_schema_version`) for the automatic schema updater to work.

Some language are missing from this pull request (`lang_de.php` and `lang_dk.php`).

I too have done those changes yesterday, but got side-tracked before opening the pull request. I'll open it right now.

**Comment by julmud** [comment]
By the way, I've just disabled the code duplication detection in SonarCloud for the language files. Sorry for the noise. :-(

File: index.php (hunk 2 above)
Comment (gjnoonan) [reply index.php:2361]: Good catch, I was going to pull this out into a helper, as it felt really dirty.

**Comment by gjnoonan** [comment]
> As there's a change of the DB schema structure, the schema version should be updated in schema.sql and schema.pre-4.1.sql (`db_schema_version`) and in global.php (`$code_schema_version`) for the automatic schema updater to work.

Good to know, thanks, I'll keep that in mind when making any more DB changes. To be honest I thought pre-4.1 would be considered legacy at the moment.

> I too have done those changes yesterday, but got side-tracked before opening the pull request. I'll open it right now.

I've approved yours, to merge in and close this one :-)

**Comment by julmud** [comment]
> To be honest I thought pre-4.1 would be considered legacy at the moment.

It is... but as long as the file is present (and the code checks it), I feel compelled to keep it up to date for completeness sake. ;-) I should probably tackle issue #9 soon and clean up this part.

**Pull request closed without merging**